### PR TITLE
docs: add `Project` to Glossary

### PR DESCRIPTION
**Pull request by klesh**
# Summary

related to apache/incubator-devlake#3468

### CODE REVIEW

**Review by ankitmalakar7 — DISMISSED** [review]
Changes looks good to me.

File: docs/Glossary.md [hunk 1]
```diff
@@ -13,6 +13,16 @@ description: >
 
 The following terms are arranged in the order of their appearance in the actual user workflow.
 
+### Projects
+**A project is a set of related [Data Scopes](#data-scope), the Data Scopes may contains different [Data Entities](#data-entities) from arbitrary [Data Sources](#data-sources) for Metric Computation**
```
Comment (Startrekzky) [inline docs/Glossary.md]: 1. Text change: "may contains" to "may contain"
2. Can you add an example to explain what is a concrete data scope/entities?  Something like "For example, a project collects the data from GitHub repo A&B and Jira board C. GitHub Repo A, GitHub repo B and Jira board C are three separate Data Scopes......"

File: docs/Glossary.md [hunk 2]
```diff
@@ -13,6 +13,16 @@ description: >
...
+
+Project was introduced in v0.15 to solve some complex metric computation problems. For example, There was a **Change Failure Rate** in [DORA Metric](../UserManuals/DORA) which requires connecting Incident(issue.type = 'INCIDENT') to its related Deployment(cicd_task.type = 'DEPLOYMENT') to determine whether a Deployment failed or not. Unfortunately, most data sources do not offer this relationship, the only reasonable approach so far is to connect Incident to its Closest Deployment, without Project,
```
Comment (Startrekzky) [inline docs/Glossary.md]: 1. "Project was introduced in v0.15" to "Project will be introduced in v0.15"
2. typo: "For example, There was" to "For example, there is"
3. "the only reasonable approach so far is to connect Incident to its Closest Deployment, without Project," to "the only reasonable approach so far is to map an Incident to its closest Deployment."
4. why is there a ", without Project," end with a `,`?

File: docs/Glossary.md [hunk 3]
```diff
@@ -13,6 +13,16 @@ description: >
...
 The following terms are arranged in the order of their appearance in the actual user workflow.
 
+### Projects
+**A project is a set of related [Data Scopes](#data-scope), the Data Scopes may contains different [Data Entities](#data-entities) from arbitrary [Data Sources](#data-sources) for Metric Computation**
+
+Project was introduced in v0.15 to solve some complex metric computation problems. For example, There was a **Change Failure Rate** in [DORA Metric](../UserManuals/DORA) which requires connecting Incident(issue.type = 'INCIDENT') to its related Deployment(cicd_task.type = 'DEPLOYMENT') to determine whether a Deployment failed or not. Unfortunately, most data sources do not offer this relationship, the only reasonable approach so far is to connect Incident to its Closest Deployment, without Project, 
+
+A project consists of the following elements:
+A set of Metrics(currently, only DORA)
+A [Blueprint](#blueprints) for active data collection
```
Comment (Startrekzky) [inline docs/Glossary.md]: 1. A [`Blueprint`](#blueprints) to pull data from [supported data sources](https://devlake.apache.org/docs/next/SupportedDataSources). You can change it to the relative path.
2. A set of `Incoming Webhooks` to receive pushed from non-supported data sources

**Review by Startrekzky — CHANGES_REQUESTED** [review]

File: docs/Glossary.md (hunk 2 above)
Comment (keon94) [reply docs/Glossary.md]: Is it expected that Incident issues always have an associated CICD Task?

Comment (klesh) [reply docs/Glossary.md]: > Is it expected that Incident issues always have an associated CICD Task?

I think so

File: docs/Glossary.md [hunk 4]
```diff
@@ -13,6 +13,18 @@ description: >
 
 The following terms are arranged in the order of their appearance in the actual user workflow.
 
+### Projects
+**A project is a set of related [Data Scopes](#data-scope), the Data Scopes may contain different [Data Entities](#data-entities) from arbitrary [Data Sources](#data-sources) for Metric Computation**
```
Comment (Startrekzky) [inline docs/Glossary.md]: Text change:
"Apache DevLake supports users to view metrics based on projects. A `project` is associated with multiple sets of [data scope](#data-scope), such as GitHub/GitLab repositories, Jira boards, Jenkins pipelines, etc. Metrics for a project are calculated based on the data entities(#data-entities) under the project's data scope.

For example, when a user associates 'Jenkins Job A' and  'Jira board B' with project 1, then ONLY `deployments` in 'Jenkins Job A' and `incidents` in 'Jira board B' will be used to calculate **Change Failure Rate** for project 1.

File: docs/Glossary.md [hunk 5]
```diff
@@ -13,6 +13,18 @@ description: >
...
+
+Project was introduced in v0.15 to solve some complex metric computation problems. For example, There is a **Change Failure Rate** in [DORA Metric](../UserManuals/DORA) which requires connecting Incident(issue.type = 'INCIDENT') to its related Deployment(`cicd_task.type = 'DEPLOYMENT'`) to determine whether a Deployment failed or not. Unfortunately, most data sources do not offer this relationship, the only reasonable approach so far is to map an Incident to its Closest Deployment. Without Project, the Incident may be mapped to unrelated Deployment in database.
```
Comment (Startrekzky) [inline docs/Glossary.md]: I put it in the last comment for your reference.

File: docs/Glossary.md [hunk 6]
```diff
@@ -13,6 +13,18 @@ description: >
...
+
+A project consists of the following elements:
+
+- A set of [Data Scopes](Glossary.md#data-scope), for example, several `repos` pointing to Github Repositories, `boards` pointing to a couple of Jira Scrum Boads and `cicd_scopes` to Jenkins Servers etc.
```
Comment (Startrekzky) [inline docs/Glossary.md]: Text change:
"A set of data scope" to "One or Multiple sets of data scope"

File: docs/Glossary.md [hunk 7]
```diff
@@ -13,6 +13,18 @@ description: >
...
+- A set of Metrics(currently, only DORA).
```
Comment (Startrekzky) [inline docs/Glossary.md]: When I read the whole sentence, I had no idea about 'A project consists the following elements: a set of metrics'. 
Do you mean that there's "a set of metrics" as a project's attribute?

File: docs/Glossary.md [hunk 8]
```diff
@@ -13,6 +13,18 @@ description: >
...
+- A [`Blueprint`](#blueprints) to pull data from [supported data sources](./SupportedDataSources.md).
+- [Optional] A set of `Incoming Webhooks` to receive pushed from non-supported data sources.
```
Comment (Startrekzky) [inline docs/Glossary.md]: - to receive pushed (data) from ... ?
- non-supported -> unsupported?

**Review by Startrekzky — CHANGES_REQUESTED** [review]

**Review by Startrekzky — APPROVED** [review]
LGTM

**Pull request merged**